KMaina/Store-Manager-API
# #161243755 Build out the REST Endpoint for Products

#### What does this PR do?
Allow a user create and fetch products for the Store Manager API 
#### Description of Task to be completed?
Have the following endpoints working

`POST /api/v1/products`

`GET /api/v1/products`

`GET /api/v1/products/<product_id>`

#### How should this be manually tested?

- clone the repo locally

- cd into the Store-Manager-API folder
- run `git pull https://github.com/KMaina/Store-Manager-API.git ft-products-enpoint-161243755`

- in a terminal run the command `pytest`

- ensure that the following environment variables are set

```
FLASK_ENV=development
FLASK_APP=run.py
```

- in a terminal run flask run to start the server

- in Postman use the following JSON body to add a product

```
{
   "name" : "Bread",
   "quantity" : 30,
   "price" : 50,
   "reorder"  : 20 
}
```
#### Any background context you want to provide?
This is all functionality for the products endpoint
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/161243755
